### ISSUE TITLE
Add ignore: ["comments"] option to max-line-length

### ISSUE THREAD

**Issue by giladgray** [stylelint/stylelint#1615]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Feature request for a new `ignore` option value for `max-line-length` rule to ignore line length _within_ comments.

Currently it supports an option to ignore _non-comments_ which honestly makes no sense to me. I would much rather ignore line length inside comments because they can easily be long (our source code includes long markdown documentation blocks) and enforce line length for the source code itself. A good example of a long comment line is one including a link to a very long URL which easily exceeds any reasonable line length limits.

> Which rule, if any, is this issue related to?

`max-line-length`

> What CSS is needed to reproduce this issue?

```css
/* @see http://lorem-ipsum-dolor-sit.amet/consectetur/adipisicing/elit/maxime/temporibus/quidem/ipsa/aspernatur-voluptas-rerum-labore-odit.omnis?dignissimos=obcaecati */
/* ^ line above is way too long but i can't break the link onto multiple lines. */
.foo {
  color: pink;
}
```

> Expected configuration to validate above source code:

```json
{
  "rules": {
    "max-line-length": ["100", {
      "ignore": ["comments"]
    }]
  }
}
```

Note: `"ignore"` option currently only supports `"non-comments"` value.

> Which version of stylelint are you using?
 
`6.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Node API + gulp plugin

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

Line in question is flagged as being way too long.

**Comment by jeddy3** [comment]
@giladgray Thanks for writing up the request so well.

This also seems like a solid addition to me. Does anyone see any issues with it, or can I label up as "help wanted"?

**Comment by ntwb** [comment]
SGTM

FYI: Previously `ignore: "non-comments"` was added to `max-line-length` in #617/#630

**Comment by jeddy3** [comment]
@giladgray Feel free to contribute this feature. As @ntwb pointed out, #630 is a PR that implements the previous ignore option for this rule. You can use that as a blueprint. You might also find the [developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md) useful for getting started.

**Comment by ntwb** [comment]
@giladgray wrote:
> "Currently it supports an option to ignore _non-comments_ which honestly makes no sense to me"

https://github.com/stylelint/stylelint/tree/master/src/rules/max-line-length#optional-options

The `max-line-length` docs for this option state:
> "Only enforce the line-length limit for lines within comments."
> 
> "This does not apply to comments that are stuck in between other stuff, only to lines that begin at the beginning or in the middle of a comment."


Maybe some tweaks to that option description would help:
> "Only enforce the line-length limit for comment blocks and line/inline comments.
>
> "This does not apply to comments that are stuck in between other stuff, only to lines that begin at the beginning or in the middle of a comment, or "shared line comments" as stylelint [proposes to name these](https://github.com/stylelint/stylelint/issues/1516#issuecomment-231598616) comments `comment-no-shared-line` where the comment is on the same line as the code.

@giladgray Do you think that better describes this option? I think @jeddy3 might be able to perform some succinct magic on that last bit to make it even better?

**Comment by giladgray** [comment]
what does "lines that begin in the middle of the comment" mean? i imagine you're talking about this:

```js
code.code(code) // comment comment
```

**Comment by ntwb** [comment]
Actually, this:

`a { color: 0; top: 0; /* too long comment here */ bottom: 0; }`

**Comment by sendilkumarn** [comment]
@ntwb  how about ignoring the comment length if it is in between ?

**Comment by gaidarenko** [comment]
May I fix this issue?

**Comment by evilebottnawi** [comment]
@giladgray let's go

**Comment by danielpost** [comment]
Related to this issue: would it be possible to add an option to ignore URLs in comments? Perhaps force URLs to be on their own line but ignore their length?

**Comment by evilebottnawi** [comment]
@danielpost better solution for this to create own plugin or use `ignorePattern`

**Comment by danielpost** [comment]
@evilebottnawi Good idea, this seems to work:

```
"max-line-length":  [
  80, {
    "ignore": ["non-comments"],
    "ignorePattern": ["/(https?://([-\\w\\.]+)+(:\\d+)?(/([\\w/_\\.]*(\\?\\S+)?)?)?)/"]
  }
],
```

**Comment by evilebottnawi** [comment]
@danielpost great! just note - you can have `file`, `ftp` schema, i am not good familiar with `url` regex, maybe your can find more simple and effectively :smile:

**Comment by danielpost** [comment]
@evilebottnawi I know, but in this case I just needed a way to place regular URLs in comments without triggering an error. If I do need to use `ftp` etc, I can always improve it.
